mozci/task.py

```diff
@@ -280,12 +280,21 @@ class GroupResult:
     duration: Optional[int]
 
 
+class FailureType(Enum):
+    TIMEOUT = "timeout"
+    CRASH = "crash"
+    GENERIC = "generic"
+
+
 @dataclass
 class TestTask(Task):
     """Subclass containing additional information only relevant to 'test' tasks."""
 
     _results: Optional[List[GroupResult]] = field(default=None)
     _errors: Optional[List] = field(default=None)
+    _failure_types: Optional[Dict[str, List[Tuple[str, FailureType]]]] = field(
```

Comment:
Nit: if you define `GroupName = NewType("GroupName", str)` and `TestName = NewType("TestName", str)` at the top of the file, you can rewrite this as `_failure_types: Optional[Dict[GroupName, List[Tuple[TestName, FailureType]]]]`, which makes it a bit clearer (I know we are not doing it in other places, but it'd be nice to start).